.github/workflows/ci.yml

```diff
@@ -153,6 +153,53 @@ jobs:
 
     - run: make rust-benchmark
 
+  emscripten:
+    name: build on emscripten
+    # only run on push to main and on release
+    if: "success() && (startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/main')"
+    needs: [test, lint]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          components: rust-src
+          target: wasm32-unknown-emscripten
+          override: true
+
+      - uses: mymindstorm/setup-emsdk@v11
+        with:
+          version: 3.1.13
+          actions-cache-folder: emsdk-cache
+
+      - name: set up python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: set package version
+        run: python .github/set_version.py
+        if: "startsWith(github.ref, 'refs/tags/')"
+
+      - name: Sync Cargo.lock
+        run: cargo update -p pydantic-core
+        if: "startsWith(github.ref, 'refs/tags/')"
+
+      - run: pip install -U --pre maturin
+
+      - name: build wheels
+        run: maturin build --release --target wasm32-unknown-emscripten --out dist -i 3.10
+
+      - run: ls -lh dist
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: wasm_wheels
```

Comment:
Since PyPI does not allow uploading wasm32 wheels right now, I choose to not putting them in `pypi_files` artifact.

https://github.com/pypa/warehouse/issues/10416

Reply:
@hoodmane do you have a link for this?

if `micropip` can install via a URL, we could upload the wasm wheel as a github release artifact and install straight from there?

I'd love to try using this package in web-assembly, but I'm not sure where to start.

Reply:
This PR added the ability to install binary wheels:
https://github.com/pyodide/pyodide/pull/2591
If you use the nightly Pyodide you can just say `await micropip.install("url_to_wheel")` and you'll be good to go, at least as long as all your binary dependencies are included in Pyodide.

Reply:
Easiest way to run tests is in node with `file:relative/path/towheel.whl` (also requires nightly Pyodide).